lib/storage/providers/SQLiteStorage.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * The SQLiteStorage provider stores everything in a key/value store by
+ * converting the value to a JSON string
+ */
+import {QuickSQLite} from 'react-native-quick-sqlite';
+import _ from 'underscore';
+
+const DB_NAME = 'Expensify-new-db';
+QuickSQLite.open(DB_NAME);
+
+QuickSQLite.execute(DB_NAME, 'CREATE TABLE IF NOT EXISTS magic_map (record_key TEXT NOT NULL PRIMARY KEY , valueJSON JSON NOT NULL) WITHOUT ROWID;');
+QuickSQLite.execute(DB_NAME, 'PRAGMA CACHE_SIZE=-20000;');
+QuickSQLite.execute(DB_NAME, 'PRAGMA SYNCHRONOUS=NORMAL;');
+QuickSQLite.execute(DB_NAME, 'PRAGMA journal_mode=WAL;');
+
+const provider = {
+    /**
+      * Get the value of a given key or return `null` if it's not available in storage
+      * @param {String} key
+      * @return {Promise<*>}
+      */
+    getItem(key) {
+        return QuickSQLite.executeAsync(DB_NAME, 'SELECT record_key, valueJSON from magic_map where record_key=?;', [key]).then(({rows}) => {
+            // eslint-disable-next-line no-underscore-dangle
+            const res = rows._array[0];
+            return JSON.parse(res.valueJSON);
+        });
+    },
+
+    /**
+      * Get multiple key-value pairs for the give array of keys in a batch
+      * @param {String[]} keys
+      * @return {Promise<Array<[key, value]>>}
+      */
+    multiGet(keys) {
+        return QuickSQLite.executeAsync(DB_NAME, `SELECT record_key, valueJSON from magic_map where record_key IN (${new Array(keys.length).fill('?').join(',')});`, keys)
+            .then(({rows}) => {
+                // eslint-disable-next-line no-underscore-dangle
+                const res = _.map(rows._array, row => [row.record_key, JSON.parse(row.valueJSON)]);
+                return res;
+            });
+    },
+
+    /**
+      * Sets the value for a given key. The only requirement is that the value should be serializable to JSON string
```

Comment:
I recall some work that we did to allow files (like images) to be stored in Onyx. I can't remember if we are actually using this feature or not, but I assume we are. And also guessing that it may not work after this change.

cc @thienlnam @kidroca to refresh our memory

Reply:
Yeah good catch, we do seem to be using this feature currently to store image in onyx while we're offline and have it upload later

https://expensify.slack.com/archives/C01GTK53T8Q/p1665588920653759

Reply:
It won't be a problem 

On native platforms images are stored on the FS while Onyx (db) only stores the path (URI) on the FS 

This happens automatically e.g. 
- pick image using the image picker
- whatever we select is resolved as `ImageSource` with `uri` key containing path to the file
- we store the `uri` in Onyx and not the actual file

Reply:
Thanks!